### ISSUE TITLE
If file name already exists.

### ISSUE THREAD

**Issue by gyke69** [DiscordDigital/ipax#13]
if file name is already exist in .JSON file dont extract its into again. 

if there is 100 ipas in a folder and you add a new ipa which will be 101th, so is it possible for ipax to check file by the name if they exist already it should not phrase it if it doesn't exist it should phrase the new ipa into the existed .JSON. 
and also if we delete delete for example 5 ipas from 101 ipa folder iPAX ipa should also check on them if they don't exist it should delete them alot with its info from that exact existin .JSON

**Comment by gyke69** [comment]
thats true that its going to overwrite there are games that are over 4GB and ipax will take much time to overwrite it.

**Comment by DiscordDigital** [comment]
Happy new year to you as well

ipax doesn't extract the full IPA file, it does only extract specific files.
Can you give me an estimated time and filesize of all IPAs it takes to regenerate the JSON file?

**Comment by gyke69** [comment]
Thank you! 
here is the result! 

98 GB files these files includes from 200 MB to 4 GB iPA files. 
took 16 minute to extract its .JSON file including app icon. 

My server info: 
contabo server 
ubuntu 20.04 
10x vCPU Cores 
60 GB RAM 
1.6 TB SSD

**Comment by DiscordDigital** [comment]
Hi Anosh

I've been wondering, the target audience of ipax is open source developers creating apps for free for everyone.
However I feel like I would never find ~100 GB of properly licensed open source iOS apps.

I'd be willing to implement dynamic changes to the JSON if you can prove that you properly have the rights for all these IPA files.

If those aren't properly licensed open source apps, I'll stick to my conclusion that ipax is fast enough for the current open source market.

**Comment by gyke69** [comment]
> Hi Anosh
> 
> I've been wondering, the target audience of ipax is open source developers creating apps for free for everyone. However I feel like I would never find ~100 GB of properly licensed open source iOS apps.
> 
> I'd be willing to implement dynamic changes to the JSON if you can prove that you properly have the rights for all these IPA files.
> 
> If those aren't properly licensed open source apps, I'll stick to my conclusion that ipax is fast enough for the current open source market.

i have got around 3 of the games pirate.
but all other application are open sourced its just i have around 8 developer accounts and i keep signing fortnite.. since its delete from app store. 

how may i prove it for ya? 
fyi - i sign fortnite with 8 develpper account. and so the other games and apps.

**Comment by DiscordDigital** [comment]
Well, since you don't have the rights to distribute Fortnite and openly admitted that there's pirated apps I can't help you with that GitHub issue.

ipax is fast enough for the current open source market.

**Comment by gyke69** [comment]
> Well, since you don't have the rights to distribute Fortnite and openly admitted that there's pirated apps I can't help you with that GitHub issue.
> 
> ipax is fast enough for the current open source market.

fortnite themselve providing people the ipa officialy in their website, so i am not sure how do you expect people to install that.

**Comment by DiscordDigital** [comment]
Because Fortnite distributes from their appstore doesn't mean you can copy Fortnite and distribute it yourself.

You need to have permission to distribute apps, even if they're freely available online, either through license or permission from the author.
